lib/ask/runtime/verboice_channel.ex

```diff
@@ -408,6 +395,28 @@ defmodule Ask.Runtime.VerboiceChannel do
     |> send_resp(200, response(hangup()) |> generate)
   end
 
+  defp notify_channel_broker(respondent, status) do
+    case respondent_channel(respondent.session) do
+      %Channel{id: channel_id} ->
+        ChannelBroker.callback_received(channel_id, respondent, status, "verboice")
+        :ok
+
+      0 ->
+        ChannelBroker.callback_received(0, respondent, status, "verboice")
+        :ok
+
+      _ ->
+        :error
+    end
+  end
+
+  defp respondent_channel(%{"current_mode" => %{"mode" => "ivr", "channel_id" => _}} = state) do
+    Ask.Runtime.Session.load_current_mode(state).channel
+  end
+
+  defp respondent_channel(%{"current_mode" => %{"mode" => "ivr"}}), do: 0
```

Comment:
We should only allow this during test where we often don't have a `channel_id` (and use `0` as a fallback). Something like (to be verified):

```suggestion
  defp respondent_channel(%{"current_mode" => %{"mode" => "ivr"}}) when Mix.env == "test", do: 0
```

Reply:
We could do the same in `Ask.Runtime.ChannelBroker` and simplify all the duplicate methods that map `nil` to `0`. I'll push another PR on Tuesday.

Reply:
@leandroradusky I pushed the commit.